src/components/CountDown/CountDown.tsx

```diff
@@ -0,0 +1,191 @@
+import {useTheme} from '../../hooks'
+
+export const FormatTime = {
+  mmss: 'mm:ss',
+  hhmmss: 'HH:mm:ss',
+  ddhhmmss: 'DD:HH:mm:ss',
+} as const
+
+export type CountDownProps = {
+  /*
+  init time countdown by second
+  */
+  initialSeconds: number
+  /*
+  call when count down finish
+  */
+  onFinish?: () => void
+  /*
+  container countdown view style
+  */
+  containerStyle?: StyleProp<ViewStyle>
+  /*
+  text countdown style
+  */
+  textStyle?: StyleProp<TextStyle>
+  /*
+  element countdown style
+  */
+  elementStyle?: StyleProp<ViewStyle>
+  /*
+  colons countdown
+  */
+  colonsStyle?: StyleProp<TextStyle>
+  /*
+  loop countdown
+  */
+  loop?: boolean
+  /*
+  format countdown as key
+  */
+  format?: keyof typeof FormatTime
+  /*
+  default count time after milisecond
+  */
+  intervalTimeBySecond?: number
+}
+const milisecond = 1000
+const numberSecondPerMinute = 60
+const secondPerHour = 3600
+const hourPerDay = 24
+const secondPerDay = 86400
+
+export const CountDown: React.FunctionComponent<CountDownProps> = ({
+  initialSeconds = 300,
+  containerStyle,
+  onFinish,
+  textStyle,
+  loop,
+  format = 'mmss',
+  intervalTimeBySecond = 1000,
+  elementStyle,
+  colonsStyle,
+}) => {
+  const CountDownTheme = useTheme().components.CountDown
+
+  const [seconds, setSeconds] = useState(initialSeconds)
+  const appState = useRef(AppState.currentState)
+  const timeEnd = useRef(new Date().getTime() + initialSeconds * milisecond)
+  const [appStateVisible, setAppStateVisible] = useState(appState.current)
+
+  /*
+    check app state is change
+  */
+  useEffect(() => {
+    const subscription = AppState.addEventListener('change', nextAppState => {
+      appState.current = nextAppState
+      setAppStateVisible(appState.current)
+    })
+    return () => {
+      subscription.remove()
+    }
+  }, [])
+
+  useEffect(() => {
+    const timeLoop = intervalTimeBySecond - new Date().getMilliseconds()
+    const timeout = setTimeout(() => {
+      if (seconds >= 0) {
+        const uff = timeEnd.current - new Date().getTime()
+        setSeconds(uff / milisecond)
+      }
+      if (seconds < 1) {
+        if (loop) {
+          timeEnd.current = timeEnd.current - new Date().getTime()
+          setSeconds(initialSeconds)
+        } else {
+          clearTimeout(timeout)
+          if (onFinish) {
+            onFinish()
+          }
+        }
+      }
+    }, timeLoop)
+    return () => {
+      clearTimeout(timeout)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [seconds, appStateVisible])
+
+  const renderColons = () => (
+    <Container>
+      <Label style={[CountDownTheme.textStyle, textStyle, colonsStyle]}>:</Label>
+    </Container>
+  )
+
+  const renderTimer = () => {
+    /*
+    caculate minutes
+    */
+    const resultCaculatMinute = Math.floor(seconds / numberSecondPerMinute) % numberSecondPerMinute
+    const minute = resultCaculatMinute >= 0 ? resultCaculatMinute : 0
```

Comment:
why it need to be `>= 0` ? shouldn't it be only `> 0`

if resultCaculateMinute = 0, it will be same as the value return from else condition